File: Haskell/08.hs
{- 
	Encontre os 10 primeiros anos bissextos.
-}
isBissexto :: Int -> Bool
isBissexto ano
    | ano `mod` 400 == 0 = True
    | ano `mod` 100 == 0 = False
    | ano `mod` 4 == 0 = True
    | otherwise = False

-- Função que gera uma lista dos primeiros 10 anos bissextos
primeirosBissextos :: [Int]
primeirosBissextos = take 10 (filter isBissexto [1..])

File: Programação Funcional/Funções/Scala/08.scala
object Main {
  // Função que verifica se um ano é bissexto
  def isBissexto(ano: Int): Boolean = {
    ano % 400 == 0 || (ano % 100 != 0 && ano % 4 == 0)
  }

  // Função principal
  def main(args: Array[String]): Unit = {
    println(isBissexto(100))
    println(isBissexto(1900))	
    println(isBissexto(2000))
    println(isBissexto(2024))	
  }
}